undofx/src/main/java/org/fxmisc/undo/UndoManagerFactory.java

```diff
@@ -14,118 +15,249 @@
+     *
+     * @param changeStream the {@link EventStream} that emits changes
+     * @param invert inverts the change
+     * @param apply applies the change (undo/redo)
+     * @param isIdentity determines whether change is an identity change (e.g. {@link Function#identity()}). For
+     *                   example, {@code 0} is the identity change in
+     *                   {@code BiFunction<Integer, Integer, Integer> plus = (i, j) -> i + j} because
+     *                   {@code 4 == 4 + 0 == plus.apply(4, 0)}
+     * @param merge merges two changes into one change. If the resulting change is an identity change, the two
+     *              changes will not be merged.
+     * @param <C> the type of change object to use
+     */
+    public static <C> UndoManager fixedSizeHistoryUndoManager(
+            EventStream<C> changeStream,
+            Function<? super C, ? extends C> invert,
+            Consumer<C> apply,
+            BiFunction<C, C, Optional<C>> merge,
+            Predicate<C> isIdentity,
+            int capacity) {
         ChangeQueue<C> queue = new FixedSizeChangeQueue<C>(capacity);
-        return new UndoManagerImpl<>(queue, invert, apply, merge, changeStream);
+        return new UndoManagerImpl<>(queue, invert, apply, merge, isIdentity, changeStream);
     }
 
+    /**
+     * Creates an {@link UndoManagerFactory} whose {@code create} methods will create an {@link UndoManager} with
+     * a limited history; when at full capacity, a new change will cause the oldest change to be forgotten.
+     */
     public static UndoManagerFactory fixedSizeHistoryFactory(int capacity) {
         return new UndoManagerFactory() {
             @Override
             public <C> UndoManager create(
                     EventStream<C> changeStream,
                     Function<? super C, ? extends C> invert,
-                    Consumer<C> apply) {
-                return fixedSizeHistoryUndoManager(changeStream, invert, apply, capacity);
-            }
-
-            @Override
-            public <C> UndoManager create(
-                    EventStream<C> changeStream,
-                    Function<? super C, ? extends C> invert,
                     Consumer<C> apply,
-                    BiFunction<C, C, Optional<C>> merge) {
-                return fixedSizeHistoryUndoManager(changeStream, invert, apply, merge, capacity);
+                    BiFunction<C, C, Optional<C>> merge,
+                    Predicate<C> isIdentity) {
+                return fixedSizeHistoryUndoManager(changeStream, invert, apply, merge, isIdentity, capacity);
             }
         };
     }
 
+    /**
+     * Creates an {@link UndoManager} with no history where all changes emitted from the EventStream will be
+     * considered non-identity changes and each emitted change will change the
+     * {@link ChangeQueue#getCurrentPosition()}; the {@link UndoManager} will never be able to undo/redo a change
+     * emitted from the changeStream.
```

Comment:
True. I added that because I wasn't sure how else to explain that `atMarkedPosition` wouldn't be the same value after a push. I guess I could just say that instead.